### PR TITLE
Provide an option to put all the Compute instances in the same AD

### PR DESCRIPTION
**Pull request by chintamani-bhat07**
In the UI show that there is an option to put all the Compute instances in the same AD.
Show that the option is not selected by default.
Screenshot 2024-12-03 at 11 15 45 AM

Show that if the option is selected in the UI that the Compute instances are all in the same AD.
Screenshot 2024-12-02 at 5 02 04 PM

IDCS instance with this option selected has the IDCS sample app added correctly. 
Screenshot 2024-12-02 at 5 57 42 PM

Show that the backends for the load balancer are set up correctly.
Screenshot 2024-12-02 at 6 00 48 PM

Perform and scale out and show that the additional instance is added in the same AD.
Screenshot 2024-12-02 at 5 50 36 PM

Show that there are no issues with cloning.
Screenshot 2024-11-29 at 11 34 23 PM
Screenshot 2024-11-29 at 11 43 21 PM
Screenshot 2024-11-29 at 11 48 23 PM

Show that there are no issues with autoscaling.

`Apply complete! Resources: 65 added, 0 changed, 0 destroyed.

Outputs:
autoscaling_function_application_id = "ocid1.fnapp.oc1.iad.aaaaaaaaazznl7uei34qyfl4sxxvvyelkmd5ufm6dw6zcw7bjpcioygt7e3q"
autoscaling_scalein_monitoring_alarm_id = "ocid1.alarm.oc1.iad.aaaaaaaajhkebpkshceyjvv6wcp64o5idne6clphnpqmtbjak5z5awuexqlq"
autoscaling_scaleout_monitoring_alarm_id = "ocid1.alarm.oc1.iad.aaaaaaaatrkvqzy3chqsipyjfckky7z6yl7xbt27t7iqp5s442a6n4nrkpna"
bastion_instance_id = "ocid1.instance.oc1.iad.anuwcljsncovviycmzj4xxabupcceaqv37c4jcx3uzjldozfhge3hpg2z34q"
bastion_instance_public_ip = "158.101.110.29"
fss_system_id = ""
fusion_middleware_control_console = ""
is_vcn_peered = false
jdk_version = "JDK 8"
listing_version = "21.3.1-210716051200"
load_balancer_id = "ocid1.loadbalancer.oc1.iad.aaaaaaaalicohve6qymbik76b62h3x6aksdd356yy2vwjdxtw3wh5rwe67ca"
load_balancer_ip = "150.136.31.236"
mount_target_id = ""
provisioning_status = "Asynchronous provisioning is enabled. Connect to each compute instance and confirm that the file /u01/data/domains/adauto_domain/provCompletedMarker exists. Details are found in the file /u01/logs/provisioning.log."
resource_identifier_value = tolist([
  "adauto-b57a7325",
  "ccbhat",
])
rms_private_endpoint_id = ""
sample_application = "https://150.136.31.236/sample-app"
sample_application_protected_by_idcs = ""
ssh_command = ""
ssh_command_with_dynamic_port_forwarding = ""
virtual_cloud_network_cidr = "10.0.0.0/16"
virtual_cloud_network_id = "ocid1.vcn.oc1.iad.amaaaaaancovviyaisnuupkdcpktyck2ygy4q2et4slpostpkrrhp7medutq"
webLogic_server_domain_configuration = "Production Mode"
weblogic_agent_configuration_id = ""
weblogic_instances = "[\""{ Instance Id:ocid1.instance.oc1.iad.anuwcljsncovviychbkobah2k27m4efwq6wjt3h2v356xuckue3xdzajc7bq, Instance name:adauto-wls-0, Availability Domain:PEKi:US-ASHBURN-AD-2, Instance Shape:VM.Standard.E4.Flex, Private IP:10.0.2.211, Public IP: }\"",\""{ Instance Id:ocid1.instance.oc1.iad.anuwcljsncovviycif6fifncrtlfebdykgvuadfam6jjz7rvatkj63mlqs3a, Instance name:adauto-wls-1, Availability Domain:PEKi:US-ASHBURN-AD-2, Instance Shape:VM.Standard.E4.Flex, Private IP:10.0.2.157, Public IP: }\""]"
weblogic_log_group_id = ""
weblogic_log_id = ""
weblogic_server_administration_console = "https://10.0.2.211:7002/console"
weblogic_version = "12.2.1.4 Enterprise Edition (Non JRF)" `

if the option is not selected in the UI that the Compute instances are still spread across ADs (regression test).
Screenshot 2024-12-02 at 6 16 45 PM

### CODE REVIEW

File: terraform/modules/compute/wls_compute/variables.tf [hunk 1]
```diff
@@ -26,6 +26,12 @@ variable "availability_domain" {
   description = "The label of the availability domain where the compute will be created"
 }
 
+variable "place_all_compute_in_same_ad" {
+  type        = bool
+  description = "Set to true if you want to use same Availability Domain for all Compute Instances."
```
Comment (HarshadVilekar) [inline terraform/modules/compute/wls_compute/variables.tf:31]: "to make a bit more clear - say 'AD of admin compute instance' "

Same comment for the description field in other .tf file.

**Comment by HarshadVilekar** [comment]
PR shows changes not related to this fix. 
- Resync the branches. 
- Are the white space changes intentional ? If not - avoid those.

File: terraform/modules/compute/wls_compute/variables.tf (hunk 1 above)
Comment (chintamani-bhat07) [reply terraform/modules/compute/wls_compute/variables.tf]: Modified the description

**Comment by chintamani-bhat07** [comment]
Updated the PR

 The white space changes are intentional to ensure proper alignment and consistent formatting.
 
 The first attachment (screenshot) shows the UI change (added a checkbox : **Use Same Availability Domain for all Compute Instances** )

**Pull request closed without merging**